lib/remap-functions/src/length.rs

```diff
@@ -0,0 +1,144 @@
+use remap::prelude::*;
+use std::convert::TryFrom;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Length;
+
+impl Function for Length {
+    fn identifier(&self) -> &'static str {
+        "length"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Array(_) | Value::Map(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(LengthFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct LengthFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for LengthFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        use Value::*;
+
+        let value = self.value.execute(state, object)?;
+
+        match value {
+            Array(v) => safe_i64(v.len()),
+            Map(v) => safe_i64(v.len()),
+            _ => Err("unsupported type".into()),
+        }
```

Comment:
Should this also work for strings? e.g. `length("foo") == 3`

(we should decide what length we want to return, I suggest just returning the length in bytes to keep it simple for the common use-case)